aspnetcore/mvc/compatibility-version/samples/2.x/CompatibilityVersionSample/Startup2.cs

```diff
@@ -5,26 +5,26 @@
 using Microsoft.AspNetCore.Mvc.Formatters;
 using Microsoft.Extensions.DependencyInjection;
 
-namespace WebApp1
+namespace CompatibilityVersionSample
```

Comment:
Next time don't change namespace. I prefer if most samples use `WebApp` - it makes it a bit easier to share code, generate a sample, etc. If the namespace isn't featured - why not standardize on `WebApp` ?
cc @scottaddie  for your thoughts.

Reply:
All of the samples have namespaces that match the template behavior ... i.e., they match whatever folder the sample is generated in. It did occur to me (a few times actually) that we could always use the same namespace. However, the pattern for upgrading these samples major release-to-major release is to copy the prior release sample (with whatever namespace it uses) and then upgrade the copy for the new release.

The main reason tho that I'm using namespaces that match the sample folder and project file name is the template behavior experience. This is what devs will see when they use the templates with `dotnet new`.

I guess it's also fair to say at this point that for consistency sake, we should stick with what I've been doing. Over half of the samps on the repo have been updated on the scheme I'm using, and it would be a :bear: to go back and update all of the samples to conform to a single namespace name.

Reply:
One more note on this ... I feel we should keep all of the following in sync ...

* The sample's folder name.
* The project file (*csproj*) file name.
* The namespace.

... also because when a sample is pulled down, we probably want readers to be able to distinguish samples locally. It would suck if someone had a handful of our samples locally but couldn't, for example, just look at the folder names (or be inside a folder looking at the project file file name) and not be able to immediately tell one sample from another one.

If we're going to have the sample folder names and project file file names be a meaningful sample name, then it stands to reason that the namespace should match. Again, it's the `dotnet new` behavior anyway.